pkg/sink/kafka/v2/factory.go

```diff
@@ -333,7 +331,6 @@ func (s *syncWriter) Close() {
 type asyncWriter struct {
 	w            Writer
 	changefeedID model.ChangeFeedID
-	closedChan   chan struct{}
```

Comment:
what is the information send to this chan before?

Reply:
to indicate the sink is closed, now it's controlled by the `context.Cancel`

Reply:
Where will the ctx be closed?

Reply:
`kafkaDMLProducer.Close()` cancel the context